include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -493,10 +493,12 @@ __radix_sort_reorder_submit(_ExecutionPolicy&& __exec, ::std::size_t __segments,
                     __offset_arr[__radix_state_idx] = __scanned_bin + __offset_rng[__local_offset_idx];
                 }
 
-                const ::std::size_t __outside_of_segment =
-                    sycl::min(__start_idx + __block_size * __blocks_per_segment, __inout_buf_size);
+                ::std::size_t __seg_end =
+                    sycl::min(__seg_start + __block_size * __blocks_per_segment, __inout_buf_size);
+                const ::std::uint32_t __residual = (__seg_end - __seg_start) % __sg_size;
+                __seg_end -= __residual;
```

Comment:
Do you think it is worth adding a comment explaining the PR here?  Something like

`// separating iterations which are not an exact multiple of the subgroup size to ensure equal visits group algs inside __peer_contribution`

Since group algorithms are abstracted inside this separate function, such a comment may be helpful for future refactors.

Reply:
Thanks; added a comment.